cmd/bucket-replication.go

```diff
@@ -3170,3 +3176,45 @@ func (p *ReplicationPool) saveStats(ctx context.Context) error {
 	}
 	return saveConfig(ctx, p.objLayer, getReplicationStatsPath(), data)
 }
+
+// getReplicationMRF returns MRF entries for a node. If node name is unspecified, return MRF for all nodes. Likewise, if bucket name is empty,
+// return MRF for all buckets.
+func (p *ReplicationPool) getReplicationMRF(ctx context.Context, objAPI ObjectLayer, bucket string, node string) (chan madmin.ReplicationMRF, error) {
+	mrfCh := make(chan madmin.ReplicationMRF, 4000)
+	go func() {
+		defer close(mrfCh)
+		for hostName := range globalRemoteEndpoints {
+			if contextCanceled(ctx) {
+				// Just consume input...
+				continue
+			}
+			if node != "all" && hostName != node {
+				continue
+			}
+			nodeNameSum := sha256.Sum256([]byte(hostName))
+			nodeNameHexEnc := hex.EncodeToString(nodeNameSum[:])
+			file := pathJoin(replicationMRFDir, nodeNameHexEnc+".bin")
```

Comment:
we should perhaps move this calculation to `globalRemoteEndpoints` so that we can avoid this repeated calculation.